walker.go

```diff
@@ -0,0 +1,84 @@
+package ftp
+
+import (
+	"fmt"
+	"strings"
+)
+
+//Walker traverses the directory tree of a remote FTP server
+type Walker struct {
+	serverConn *ServerConn
+	root       string
+	cur        item
+	stack      []item
+	descend    bool
```

Comment:
It feels like the boolean should be on the item, as it is linked to the folder

Reply:
Again, I was aiming for parity between this and the Walker implementation from the standard library where the descend is also on the Walker struct. This allows the caller to only have to use the Walker object to traverse the directory tree whilst maintaining a clear separation between the caller and the package.